pkg/config/shared.go

```diff
@@ -549,6 +549,11 @@ func (f MQTTConfigProviderFunc) GetMQTTConfig(ctx context.Context) (*MQTT, error
 	return f(ctx)
 }
 
+// RateLimitingRedis represents configuration for the in-Redis rate limiting store.
+type RateLimitingRedis struct {
+	Config redis.Config `name:"config"`
```

Comment:
This still duplicates the Redis configuration on per rate limiter basis.

Please take a look at how [stores](https://github.com/TheThingsNetwork/lorawan-stack/blob/012d0d9706514b5393fa54c2168e4eda31feb55c/pkg/applicationserver/config.go#L110) are propagated from the `start` command to the actual implementations.

Compared to a store, you won't have an interface, because the implementation is done by another package, but you can pass a `*ttnredis.Client` and it will work just fine.